src/cunumeric/random/bitgenerator.cc

```diff
@@ -32,6 +32,15 @@ static Legion::Logger log_curand("cunumeric.random");
 
 Legion::Logger& randutil_log() { return log_curand; }
 
+void randutil_check_curand(curandStatus_t error, const char* file, int line)
+{
+  if (error != CURAND_STATUS_SUCCESS) {
+    randutil_log().fatal() << "Internal CURAND failure with error " << (int)error << " in file "
```

Comment:
Maybe better to do `static_cast<int>(error)` ?

Reply:
let's not make any quality improving changes here. I'm planning a couple of follow-up PRs after this.